bika/lims/utils/__init__.py

```diff
@@ -382,31 +389,41 @@ def isnumber(s):
         return False
 
 
-def bika_url_fetcher(url):
+def senaite_url_fetcher(url):
     """Basically the same as the default_url_fetcher from WeasyPrint,
     but injects the __ac cookie to make an authenticated request to the resource.
     """
-    from weasyprint import VERSION_STRING
-    from weasyprint.compat import Request
-    from weasyprint.compat import urlopen_contenttype
 
+    logger.info("Fetching URL '{}' for WeasyPrint".format(url))
+
+    # get the pyhsical path from the URL
     request = api.get_request()
-    __ac = request.cookies.get("__ac", "")
+    path = "/".join(request.physicalPathFromURL(url))
+
+    # fetch the object by sub-request
+    portal = api.get_portal()
+    context = portal.restrictedTraverse(path, None)
+
+    if context is None:
+        logger.info("URL is external, passing over to the default URL fetcher...")
+        return default_url_fetcher(url)
+
+    logger.info("URL is local, fetching data by path '{}' via subrequest".format(path))
+
+    # get the data via subrequest
+    response = subrequest(path)
```

Comment:
Here I'm not sure if I should hand in the URL directly or the physical path. I didn't yet analyse the code from p.s. to see if this makes a change at all. If not, we could probably remove all the URL-Path conversion from above...

Reply:
As far as I see in the examples from https://pypi.python.org/pypi/plone.subrequest , they are always using the path, never the full URL. But well, we can inspect that `subrequest` function

Reply:
Maybe you are right, see this:
https://github.com/plone/plone.subrequest/blob/8b5cd141b3a2adc3d1dad381376c7e5adc80da22/plone/subrequest/__init__.py#L103-L114